tests/integration/ha/test_ha.py

```diff
@@ -37,8 +75,83 @@ async def test_build_and_deploy(ops_test: OpsTest, kafka_charm):
         assert ops_test.model.applications[APP_NAME].status == "active"
         assert ops_test.model.applications[ZK_NAME].status == "active"
 
+    await ops_test.model.add_relation(APP_NAME, f"{DUMMY_NAME}:{REL_NAME_ADMIN}")
+    await ops_test.model.wait_for_idle(apps=[APP_NAME, DUMMY_NAME], idle_period=30)
+    assert ops_test.model.applications[APP_NAME].status == "active"
+    assert ops_test.model.applications[DUMMY_NAME].status == "active"
+
+
+async def test_replicated_events(ops_test: OpsTest):
+    await ops_test.model.applications[APP_NAME].add_units(count=2)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", timeout=600, idle_period=120, wait_for_exact_units=3
+    )
+    logger.info("Producing messages and checking on all units")
+    produce_and_check_logs(
+        model_full_name=ops_test.model_full_name,
+        kafka_unit_name=f"{APP_NAME}/0",
+        provider_unit_name=f"{DUMMY_NAME}/0",
+        topic="replicated-topic",
+        replication_factor=3,
+        num_partitions=1,
+    )
+    # check logs in the two remaining units
+    check_logs(
+        model_full_name=ops_test.model_full_name,
+        kafka_unit_name=f"{APP_NAME}/1",
+        topic="replicated-topic",
+    )
+    assert await get_topic_offsets(
+        ops_test=ops_test, topic="replicated-topic", unit_name=f"{APP_NAME}/1"
+    ) == ["0", "15"]
```

Comment:
**niptick** as mentioned in another comment, I'd make 15 as a parameter that is passed to "produce_and_check_logs" and also here. This should make things easier also for who reads the tests